tt_eager/tt_dnn/op_library/eltwise_unary/eltwise_unary_op.hpp

```diff
@@ -137,27 +139,28 @@ struct EltwiseUnary {
 Tensor eltwise_unary(const EltwiseUnary& op, const Tensor& input_tensor);
 
 operation::ProgramWithCallbacks eltwise_unary_multi_core_height_or_block_sharded(
-    const Tensor& a, Tensor& output, const std::vector<UnaryWithParam> op_chain);
+    const Tensor& a, Tensor& output, const std::vector<UnaryWithParam> op_chain, bool fp32_dest_acc_en);
 operation::ProgramWithCallbacks eltwise_unary_multi_core(
-    const Tensor& a, Tensor& output, const std::vector<UnaryWithParam> op_chain);
+    const Tensor& a, Tensor& output, const std::vector<UnaryWithParam> op_chain, bool fp32_dest_acc_en);
 operation::ProgramWithCallbacks eltwise_unary_single_core(
-    const Tensor& a, Tensor& output, const std::vector<UnaryWithParam> op_chain);
+    const Tensor& a, Tensor& output, const std::vector<UnaryWithParam> op_chain, bool fp32_dest_acc_en);
 
 inline Tensor run_eltwise_unary(
     const Tensor& input_tensor,
     std::vector<UnaryWithParam> ops_chain,
     const MemoryConfig& output_mem_config = operation::DEFAULT_OUTPUT_MEMORY_CONFIG) {
     TT_FATAL(ops_chain.size() > 0, "At least 1 unary op must be specified");
+    bool fp32_dest_acc_en = input_tensor.get_dtype() == DataType::UINT32;       // MT: Currently only uint32 is moved to DST directly, fp32 is converted to fp16b
```

Comment:
can float32 and int32 be made to do this unpack to dest directly as well, right? Is the problem just how to pipe that config down to the op?

Reply:
Yes, float32 and int32 should and will be part of this, but at this point float32 can't be preserved from unpacker's point of view do to some missing changes how in the `data_format.cpp`. Currently, fp32 as unpacker src format is recognized and unpacker dst format is being selected automatically as float16b, which is OK for Grayskull but not for wormhole. We need an additional parameter in the compile flow that will be enabled only for Wormhole and that will "tell" the compiler to preserve fp32. I will attend to this later as no one has requested it yet, uint32 is more important and it works.
Int32 should be part of this as well, but it's currently not supported as ttnn format.

Reply:
Great, thanks! fyi @davorchap